tests/admin_views/tests.py

```diff
@@ -1119,9 +1119,30 @@ def setUp(self):
     def test_save_as_duplication(self):
         """Ensure save as actually creates a new person"""
         post_data = {'_saveasnew': '', 'name': 'John M', 'gender': 1, 'age': 42}
-        self.client.post(reverse('admin:admin_views_person_change', args=(self.per1.pk,)), post_data)
+        response = self.client.post(reverse('admin:admin_views_person_change', args=(self.per1.pk,)), post_data)
         self.assertEqual(len(Person.objects.filter(name='John M')), 1)
         self.assertEqual(len(Person.objects.filter(id=self.per1.pk)), 1)
+        new_person = Person.objects.latest('id')
+        self.assertRedirects(response, reverse('admin:admin_views_person_change', args=(new_person.pk,)))
+
+    def test_save_as_continue_false(self):
+        """
+        Ensure saving a new object using "Save as new" redirects to changelist
+        instead of change view when the `save_as_continue` attribute is set to
+        `False`.
+        """
+        class CustomPersonAdmin(ModelAdmin):
+            save_as_continue = False
+
+        site.unregister(Person)
```

Comment:
I see, thanks for the help. Is something like `site3` along the other ones a proper way to do that? In your example you reuse the existing `site2` one, right?